flagsmith-jira-app/src/frontend/components/IssueFeatureTables.tsx

```diff
@@ -148,30 +146,64 @@ const IssueFeatureTable = ({
   const featureId = environmentFeatures[0]!.id;
   const featureName = environmentFeatures[0]!.name;
 
+  const projectUrl = `${FLAGSMITH_APP}/project/${environmentFeatures[0]?.project}`;
+
   /** Read feature state for each environment */
-  const readFeatureState = useCallback(
-    async (): Promise<FeatureState> => ({
-      featureId,
-      environments: await Promise.all(
-        environments.map(async (environment) => ({
-          ...(await readEnvironmentFeatureState({
+  const readFeatureState = useCallback(async (): Promise<FeatureState> => {
+    const featureProjectId = environmentFeatures[0]?.project;
+
+    const matchingEnvs = environments.filter(
+      (env) => String(env.project) === String(featureProjectId),
+    );
+
+    const environmentFeatureStates = await Promise.all(
+      matchingEnvs.map(async (environment) => {
+        try {
+          const state = await readEnvironmentFeatureState({
             envApiKey: String(environment.api_key),
             featureName,
-          })),
-          name: environment.name,
-          api_key: String(environment.api_key),
-        })),
-      ),
-      counts: environmentFeatures.map((feature) => ({
+          });
+
+          return {
+            ...state,
+            name: environment.name,
+            api_key: String(environment.api_key),
+          };
+        } catch (err) {
+          return null;
+        }
+      }),
+    );
+
+    const validStates = environmentFeatureStates.filter(Boolean);
+
+    const relevantFeatures = environmentFeatures.filter(
+      (f) => String(f.project) === String(featureProjectId),
+    );
+
+    return {
+      featureId,
+      environments: validStates as EnvironmentFeatureState[],
+      counts: relevantFeatures.map((feature) => ({
         variations: feature.multivariate_options.length,
         segments: feature.num_segment_overrides ?? 0,
         identities: feature.num_identity_overrides ?? 0,
       })),
-    }),
-    [environments, environmentFeatures],
-  );
+    };
+  }, [environments, environmentFeatures]);
+
   const [state] = usePromise(
-    async () => (error === undefined ? readFeatureState() : undefined),
+    async () => {
+      if (error === undefined) {
+        try {
+          return await readFeatureState();
+        } catch (err) {
+          console.error("[ERROR] Failed to read feature state:", err);
```

Comment:
Is this needed if it's thrown? I think `ErrorWrapper` will call `console.error` anyway.